admin/module/catalog-admin-module-sources/src/main/java/org/codice/ddf/catalog/admin/plugin/AdminSourcePollerServiceBean.java

```diff
@@ -0,0 +1,242 @@
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.ddf.catalog.admin.plugin;
+
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.management.InstanceAlreadyExistsException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+
+import org.codice.ddf.ui.admin.api.ConfigurationAdminExt;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.metatype.ObjectClassDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.service.ConfiguredService;
+import ddf.catalog.source.ConnectedSource;
+import ddf.catalog.source.FederatedSource;
+import ddf.catalog.source.Source;
+
+public class AdminSourcePollerServiceBean implements AdminSourcePollerServiceBeanMBean {
+    static final String META_TYPE_NAME = "org.osgi.service.metatype.MetaTypeService";
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(AdminSourcePollerServiceBean.class);
+
+    private static final String MAP_ENTRY_ID = "id";
+
+    private static final String MAP_ENTRY_ENABLED = "enabled";
+
+    private static final String MAP_ENTRY_FPID = "fpid";
+
+    private static final String MAP_ENTRY_NAME = "name";
+
+    private static final String MAP_ENTRY_BUNDLE_NAME = "bundle_name";
+
+    private static final String MAP_ENTRY_BUNDLE_LOCATION = "bundle_location";
+
+    private static final String MAP_ENTRY_BUNDLE = "bundle";
+
+    private static final String MAP_ENTRY_PROPERTIES = "properties";
+
+    private static final String MAP_ENTRY_CONFIGURATIONS = "configurations";
+
+    private static final String DISABLED = "_disabled";
+
+    private static final String SERVICE_NAME = ":service=admin-source-poller-service";
+
+    private final ObjectName objectName;
+
+    private final MBeanServer mBeanServer;
+
+    private final BundleContext bundleContext;
+
+    private final ConfigurationAdmin configurationAdmin;
+
+    public AdminSourcePollerServiceBean(ConfigurationAdmin configurationAdmin) {
+        bundleContext = getBundleContext();
```

Comment:
You cannot cache the bundleContext.  It can go stale and throw exceptions.  You have to call getBundleContext everything you want to use it.